lxd/device/gpu_physical.go

```diff
@@ -102,6 +103,35 @@ func (d *gpuPhysical) Start() (*deviceConfig.RunConfig, error) {
 // Returns RunConfig populated with mount info required to pass the unix-char devices into the container.
 func (d *gpuPhysical) startContainer() (*deviceConfig.RunConfig, error) {
 	runConf := deviceConfig.RunConfig{}
+	if d.config["id"] != "" {
+		// Check if the id of the device match a CDI format.
+		cdiID, err := cdi.ToCDI(d.config["id"])
```

Comment:
In other integrations such as the docker CLI we use
```
		if cdiparser.IsQualifiedName(device) {
			cdiDeviceNames = append(cdiDeviceNames, device)
			continue
		}
```

See for example https://github.com/docker/cli/blob/ffb842597a9d060931c7d4c9bf6b21afbccf5440/cli/command/container/opts.go#L480-L483

Reply:
@gabrielmougard have you considered @elezar suggestion here?

Reply:
The `ToCDI` already calls `cdiparser.IsQualifiedName` internally and will give a valid cdiID only if this is true. It is also possible to have an empty cdiID and a nil error, which mean that `cdiparser.IsQualifiedName` is false but that the id can the a DRM card ID that is a valid use case for us. The only time where we can have a non-nil error is if `cdiparser.IsQualifiedName` is true but the values (like vendor or class of the fq CDI name) are not supported by LXD (e.g an `amd.com` vendor)